src/daemon/org.projectatomic.rpmostree1.xml

```diff
@@ -278,6 +278,18 @@
       <arg type="s" name="transaction_address" direction="out"/>
     </method>
 
+   <!-- Available options:
+        "editor" (type 'b')
+        "merge" (type 'b')
+    -->
+    <method name="KernelArgs">
+      <arg type="as" name="kernel_args_added" direction="in"/>
+      <arg type="as" name="kernel_args_replaced" direction="in"/>
+      <arg type="as" name="kernel_args_removed" direction="in"/>
```

Comment:
The semantics here get interesting because kernel arguments are [a multimap](https://en.wikipedia.org/wiki/Multimap).   For example dracut parses `rd.lvm.lv=foo/bar rd.lvm.lv=baz/moo` etc.  So if I specify `rd.lvm.lv=hello/world` in `kernel_args_replaced`, does that eliminate all of the other `rd.lvm.lv` args?

Reply:
There's also a similar discussion for ostree remotes in https://github.com/ostreedev/ostree/pull/1166#issuecomment-332010464

Reply:
>So if I specify rd.lvm.lv=hello/world in kernel_args_replaced, does that eliminate all of the other rd.lvm.lv args

I believe so, because my original plan for backend logic was to reuse the parsing functions for OstreeKernelArgs( `_ostree_kernel_args_append_kargv` and `_ostree_kernel_args_replace_kargv` ) to parse all of the kernel arguments. And the old kernel arg parsing logic uses a  hashtable to replace the duplicate keys, so other rd.lvm.lv args existed earlier will be removed. 

This is the original code https://github.com/ostreedev/ostree/blob/master/src/ostree/ot-admin-instutil-builtin-set-kargs.c#L97 and https://github.com/ostreedev/ostree/blob/master/src/libostree/ostree-kernel-args.c#L100

Do we want that?

Reply:
**tl;dr**: Realized `_ostree_kernel_args_append_kargv()` might not be usable

After a bit more look into the code base, I realized some of the code in ostree code base does not function exactly as I thought T_T. Sorry for that and  kinda take back some of what I said earlier. Below is their logic for appending_kargs().
https://github.com/ostreedev/ostree/blob/master/src/libostree/ostree-kernel-args.c#L131

>So if I specify rd.lvm.lv=hello/world in kernel_args_replaced, does that eliminate all of the other rd.lvm.lv args

If we were to adopt the `_ostree_kernel_args_append_kargv()` , `rd.lvm.lv=hello/world` will instead be the key, and `hello/world` will be the value. So, before.. if we do a replacement by calling `_ostree_kernel_args_replace_take() `. Other `rd.lvm.lv` values will not be affected because they do not have the exact string...............

So, I am thinking I possibly need to reformat the `_ostree_kernel_args_append` function because.. having `key=value` string as a key for the KARGS hashtable seems not right to me. But, I still think replace function in ostree end should be usable. Thoughts ? And thanks for the review comments. :)